cpu/cortexm_common/thread_arch.c

```diff
@@ -332,11 +327,14 @@ __attribute__((naked)) void arch_context_switch(void)
     "ldr    r1, =sched_active_thread  \n" /* load address of current tcb */
     "ldr    r1, [r1]                  \n" /* dereference pdc */
     "str    r0, [r1]                  \n" /* write r0 to pdc->sp */
+    );
+}
+
+void __attribute__((naked)) __attribute__((used)) isr_svc(void) {
```

Comment:
Is it not dangerous to assume that these two functions will be placed in order and next to each other without any padding in the final binary? I'm sensing that we are relying on linker implementation specific behaviour here